src/helpers.rs

```diff
@@ -0,0 +1,6 @@
+use crate::error::*;
+
+pub fn split_type_name(name: &str) -> ParseResult<(&str, &str)> {
+    let index = name.rfind('.').ok_or_else(|| ParseError::InvalidTypeName)?;
+    Ok((name.get(0..index).unwrap(), name.get(index + 1..).unwrap()))
```

Comment:
nit: if you're just going to `unwrap` no need to use `get`. You can do `name[0..index]`

Reply:
Right, forgot about that. 👍